### PR TITLE
How to set up a Node.js Express server for React 

### PR DESCRIPTION
**Pull request by verah-tech**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by verah-tech** [comment]
This solves #1868

**Comment by lalith1403** [comment]
@cyberShaw I would like to review this article. I see you have a couple of articles on your plate. Thank you.

cc: @hectorkambow

**Comment by cyberShaw** [comment]
> @cyberShaw I would like to review this article. I see you have a couple of articles on your plate. Thank you.
> 
> cc: @hectorkambow

Sure thing, @lalith1403 
Go ahead :D

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by lalith1403** [comment]
This is a very good detailed and easy to follow article. Congratulations on a well written article. You will have to include a headshot pic and short bio in a folder named after your name in the authors folder. Check the authors folder for more details. 

The code is explained well and runs fine.

File: articles/how-to-setup-nodejs-express-for-react/index.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+```bash
+$cd demoreactapp
+$npx create-react-app client
+```
+
+The above commands will create a react app with the name `client` inside the root directory.
+
+### Setting up the Express Server
+Next will be to create an Express server inside the server.js. Browse to the root directory from the terminal and run the below command:
```
Comment (lalith1403) [inline articles/how-to-setup-nodejs-express-for-react/index.md:27]:
```suggestion
The next step involves creating an Express server in the file `server.js`. Browse to the root directory from the terminal and run the command mentioned below.
```

File: articles/how-to-setup-nodejs-express-for-react/index.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+Line 1 and 2 - requires Express and allows us to use it inside our `server.js` file.
+
+Line 3 – It will set the Express server on which port will run on.
+
+Line 6 – will display a message on the console that the server is working as expected.
+
+Line 9 to 11 – It will set a `GET route` that we will, later on, fetch from our client-side React App.
+
+### Setting up the proxy
+In this step, the Webpack development server was auto-generated when we ran the `create-react-app` command.  Our react App is running on the Webpack development server on the frontend.  We can proxy the API requests from the client-side to the API on the server-side. The API  on the server-side(Express server) will be running on port 5000.
```
Comment (lalith1403) [inline articles/how-to-setup-nodejs-express-for-react/index.md:64]: Explain webpack development server in short.

File: articles/how-to-setup-nodejs-express-for-react/index.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+export default App;
+```
+
+Inside of `componentDidMount()`, the function `callBackendAPI()` is called. This function will fetch the route previously created in the Express server and then `set{data: null}` to the fetch request-response.
+On the line that contains `this.state.data` renders new data  to the [DOM](https://eloquentjavascript.net/14_dom.html)
+
+### Running the App
+First will be to navigate to root directory `demoreactapp` and run the below command:
```
Comment (lalith1403) [inline articles/how-to-setup-nodejs-express-for-react/index.md:147]: Rewrite this sentence. We navigate to the root directory `demoreactapp` and run the command below:

File: articles/how-to-setup-nodejs-express-for-react/index.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+```bash
+$cd demoreactapp
+$node server.js
+```
+
+After running the `server.js`, next will be to browse to “<http://localhost:5000/express_backend>”, and the below message will be displayed:
```
Comment (lalith1403) [inline articles/how-to-setup-nodejs-express-for-react/index.md:154]: Mention file before mentioning file name. After running the file server.js

File: articles/how-to-setup-nodejs-express-for-react/index.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+$npm start
+```
+
+The above commands will run the React Dev server, which runs on port 3000, and automatically open in the web browser.
+
+The screen below with a message will be displayed:
+
+![React app running](/engineering-education/how-to-setup-nodejs-express-for-react/react-app-success.PNG)
+
+Finally, we have rendered the data fetched from the `GET route` in `server.js` to our frontend React App, as demonstrated above. If the Express server is disconnected, the React server will continue running, but the connectivity to the backend will be lost, and nothing will be displayed.
```
Comment (lalith1403) [inline articles/how-to-setup-nodejs-express-for-react/index.md:173]: Last sentence needs to be broken down into 2 simpler sentences.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Minor changes are required. Explain webpacks to add more content to the article. Overall, terrific job.

**Comment by verah-tech** [comment]
@lalith1403 The requested changes have been done. Also, the Author's information has been updated.
Thank you

**Comment by lalith1403** [comment]
The headshot image is missing in the authors folder.

**Review by lalith1403 — APPROVED** [review]

**Comment by verah-tech** [comment]
@lalith1403 headshot updated

**Comment by hectorkambow** [comment]
great work and article @verah-tech 

cc @lalith1403  🚀

**Comment by ephnjor2021** [comment]
@hectorkambow Thank you

**Comment by verah-tech** [comment]
@hectorkambow Thank you

**Pull request merged**